File: ascii-video-processor/src/app/schema.ts
export interface ASCIIConfig {
    font_size: number;
    font_path: string;
    background_color: [number, number, number];
    characters: ASCIICharacter[];
    original_color: boolean;
    rendering_method: RenderingMethod;
}

export interface ASCIICharacter {
    char: string;
    threshold: [number, number];
    color: [number, number, number];
}

export enum RenderingMethod {
    LUMINANCE = "LUMINANCE",
    EDGE_DETECTION = "EDGE_DETECTION"
}

File: ascii-video-processor/backend/src/video/schema.py
from pydantic import BaseModel
from enum import Enum
from typing import Literal


class ASCII_Character(BaseModel):
    char: str
    threshold: tuple[float, float]
    color: tuple[int, int, int]

class RenderingConfig(BaseModel):
    font_size: int
    font_path: str
    background_color: tuple[int, int, int]
    characters: list[ASCII_Character]
    original_color: bool
    rendering_method: Literal["LUMINANCE", "EDGE_DETECTION"]
    


    
    


